src/main/java/Main.java

```diff
@@ -1,71 +1,100 @@
+import Exceptions.DuplicatePersonException;
+import Inventory.*;
+import Misc.Address;
+import Misc.Insurance;
+import Person.Patient;
+import Person.Pharmacist;
+import Person.PharmacyTechnician;
+import Pharmacy.Pharmacy;
+import Product.Item;
+import Product.Medication;
+import Product.Prescription;
+import java.util.Scanner;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
 public class Main {
 
-    public static void main(String[] args) {
-        Address[] addresses = Address.predefinedAddresses();
-        Pharmacist[] pharmacists = Pharmacist.predefinedPharmacist();
-        PharmacyTechnician[] technicians = PharmacyTechnician.predefinedPharmacyTechnicians();
-        Pharmacy pharmacy = new Pharmacy("Joffrey's Pharmacy", addresses[0], "123-321-4567",
-            "pharmEmail@email.com");
-        pharmacy.setPrescriptionRegistry(new PrescriptionRegistry());
-        PrescriptionRegistry prescriptionRegistry = pharmacy.getPrescriptionRegistry();
-        ProductInventory inventory = new ProductInventory();
+    private static final Logger LOG = LogManager.getLogger(Main.class);
+
+    private static final Address[] ADDRESSES = Address.predefinedAddresses();
+    private static final Pharmacist[] PHARMACISTS = Pharmacist.predefinedPharmacist();
+    private static final PharmacyTechnician[] TECHNICIANS = PharmacyTechnician.predefinedPharmacyTechnicians();
+    private static final Pharmacy PHARMACY = new Pharmacy("PharmacyRx",
+        ADDRESSES[0], "123-321-4567", "PHARMEMAIL@EMAIL.COM");
+    private static final Prescription[] PRESCRIPTIONS = Prescription.predefinedPrescriptions();
+    private static final Medication[] MEDICATIONS = Medication.predefinedMedications();
+    private static final Patient[] PATIENTS = Patient.predefinedPatients();
+    private static final Insurance[] INSURANCES = Insurance.predefinedInsurance();
+    private static final ProductInventory PRODUCT_INVENTORY = new ProductInventory();
```

Comment:
I was thinking the same thing, especially for PRODUCT_INVENTORY since it is not "Hard coded". But Intellij recommended that this variable be marked final since they do not get reassigned (guessing), and since they are final then they should be capitalized-snake-cased. All the other capitalized variables are essentially predefined and hard-coded instances so I thought it was appropriate to capitalize. What changes do you suppose I should change in the naming?

Reply:
The DataProvider method makes a lot of sense and is more cleaner too. Added the updates. Thanks!